src/javascripts/ng-admin/Main/component/directive/maNgAdmin.js

```diff
@@ -0,0 +1,18 @@
+function maNgAdminDirective(NgAdminConfiguration) {
+    'use strict';
+
+    return {
+        scope: {
+            'configuration': '&'
```

Comment:
Well, it's not used at all since you override it with the `NgAdminConfiguration` value. And that's a problem. I would very much like to write

```html
<ma-ng-admin configuration="myCustomValue"></ma-ng-admin>
```

But even after modifying line 10 to use `configuration` from scope as default value, it doesn't work because you depend on the hardcoded `NgAdminConfiguration` value everywhere in the code (and especially in routing).

You have to find a way for this to work without any global value, just using the current scope. That means that the initial state should depend on the current scope.

Note: I've been looking for a solution for this problem for the past hour and couldn't find one. It's not an easy problem. Good luck.